### DISCUSSION TITLE
How to use company-yasnippet before company-ispell?

### DISCUSSION THREAD

**Discussion by zhenhua-wang** [company-mode/company-mode#1453] [Q&A] [answered]
I would like to get completion from `company-yasnippet` first. If no completion find, then switch to `company-ispell`. However, with the following config, `company-ispell` is completely blocked by `company-yasnippet`. Is there any way to achieve this? Thanks for any help.

```elisp
(setq company-backends '(company-files
         company-capf
         company-yasnippet
         company-ispell))
```

**Comment by dgutov** [comment, marked as answer]
There are basically two ways to do that:

* Write your own backend which will be a simple wrapper that calls both backends and returns the completions from the one that has them.
* Add an advice to `company-yasnippet` to add special handling to the `prefix` method which would call the logic in `candidates` right away, and if nothing's found, return `nil` from `prefix` already.

You could also try using `M-x company-other-backend`, which is a command to be used explicitly (through a key binding of your choice) that iterates through the backends in the list and will make sure you get to the one that returns some completions.
